src/devices/src/virtio/block/io/async_io.rs

```diff
@@ -33,6 +33,35 @@ pub struct AsyncFileEngine<T> {
     phantom: PhantomData<T>,
 }
 
+pub struct WrappedUserData<T> {
+    addr: Option<GuestAddress>,
+    user_data: T,
+}
+
+impl<T> WrappedUserData<T> {
+    fn new(user_data: T) -> Self {
+        WrappedUserData {
+            addr: None,
+            user_data,
+        }
+    }
+
+    fn new_with_addr(addr: GuestAddress, user_data: T) -> Self {
+        WrappedUserData {
+            addr: Some(addr),
+            user_data,
+        }
+    }
+
+    fn into_user_data(self, mem: &GuestMemoryMmap, count: u32) -> T {
```

Comment:
I don't find this function's name explicit. I would use something like: `mark_user_data_dirty()` or something similar. I don't have a strong preference for this particular name, so I am open to other suggestions if you have ideas.

Reply:
Makes sense. I renamed it to `mark_dirty_mem_and_unwrap()`. WDYT ?